invoices/invoiceregistry.go

```diff
@@ -932,8 +932,11 @@ func (i *InvoiceRegistry) notifyExitHopHtlcLocked(
 	}
 }
 
-// SettleHodlInvoice sets the preimage of a hodl invoice.
-func (i *InvoiceRegistry) SettleHodlInvoice(preimage lntypes.Preimage) error {
+// SettleHodlInvoice settles a hold invoice. If the preimage isn't known yet, it
+// can be passed in to this call.
+func (i *InvoiceRegistry) SettleHodlInvoice(hash lntypes.Hash,
+	preimage *lntypes.Preimage) error {
+
```

Comment:
should add a consistency check that if a preimage is supplied that it always matches the payment

Reply:
Consistency check is present in `channeldb.updateInvoiceState`